src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1243,7 +1243,7 @@ protected override BoundExpression VisitExpressionWithoutStackGuard(BoundExpress
             // Verify Visit method set _result.
             TypeSymbolWithAnnotations resultType = _resultType;
             Debug.Assert((object)resultType.TypeSymbol != _invalidType.TypeSymbol);
-            Debug.Assert(AreCloseEnough(resultType.TypeSymbol, node.Type));
+            Debug.Assert(node.Kind == BoundKind.ThrowExpression || AreCloseEnough(resultType.TypeSymbol, node.Type));
```

Comment:
> Do you think that's better?

In this visitor, after visiting each expression, we should set ```_resultType``` the result type should reflect the original type of the node plus added nullability information. It shouldn't matter if there is a conversion on top of the BoundThrowExpression node, it will be handled when its turn comes and the ```_resultType``` will be adjusted for it as appropriate. That said, there should be no special handling/treatment for BoundThrowExpression, but adjusting the assert adds it.  

---
In reply to: [253510940](https://github.com/dotnet/roslyn/pull/32991#discussion_r253510940) [](ancestors = 253510940)